service/gcs/bridge/bridge_handler_test.go

```diff
@@ -0,0 +1,800 @@
+	tb := &Bridge{coreint: mc}
+	tb.resizeConsole(rw, req)
+
+	verifyResponseSuccess(t, rw)
+	if uint32(mc.LastResizeConsole.Pid) != r.ProcessID {
+		t.Fatal("last resize console process pid was not the same")
+	}
+	if mc.LastResizeConsole.Width != r.Width {
+		t.Fatal("last resize console process width was not the same")
+	}
+	if mc.LastResizeConsole.Height != r.Height {
+		t.Fatal("last resize console process height was not the same")
+	}
+}
+
+func Test_ModifySettings_InvalidJson_Failure(t *testing.T) {
+	req := createRequest(t, prot.ComputeSystemModifySettingsV1, nil)
+	rw := createResponseWriter(req)
+
+	tb := new(Bridge)
+	tb.modifySettings(rw, req)
+
+	verifyResponseJSONError(t, rw)
+}
+
+func Test_ModifySettings_VirtualDisk_InvalidSettingsJson_Failure(t *testing.T) {
+	r := &prot.ContainerModifySettings{
+		MessageBase: &prot.MessageBase{
+			ContainerID: "test",
+			ActivityID:  "act",
+		},
+		Request: prot.ResourceModificationRequestResponse{
+			ResourceType: prot.PtMappedVirtualDisk,
+		},
+	}
+
+	req := createRequest(t, prot.ComputeSystemModifySettingsV1, r)
+	rw := createResponseWriter(req)
+
+	tb := new(Bridge)
+	tb.modifySettings(rw, req)
+
+	verifyResponseJSONError(t, rw)
+}
+
+func Test_ModifySettings_MappedDirectory_InvalidSettingsJson_Failure(t *testing.T) {
+	r := &prot.ContainerModifySettings{
+		MessageBase: &prot.MessageBase{
+			ContainerID: "test",
+			ActivityID:  "act",
+		},
+		Request: prot.ResourceModificationRequestResponse{
+			ResourceType: prot.PtMappedDirectory,
+		},
+	}
+
+	req := createRequest(t, prot.ComputeSystemModifySettingsV1, r)
+	rw := createResponseWriter(req)
+
+	tb := new(Bridge)
+	tb.modifySettings(rw, req)
+
+	verifyResponseJSONError(t, rw)
+}
+
+func Test_ModifySettings_CoreFails_Failure(t *testing.T) {
+	r := &prot.ContainerModifySettings{
+		MessageBase: &prot.MessageBase{
+			ContainerID: "test",
+			ActivityID:  "act",
+		},
+		Request: prot.ResourceModificationRequestResponse{
+			ResourceType: prot.PtMappedDirectory,
+			Settings:     &prot.MappedDirectory{}, // Default values.
+		},
+	}
+
+	req := createRequest(t, prot.ComputeSystemModifySettingsV1, r)
+	rw := createResponseWriter(req)
+
+	tb := &Bridge{
+		coreint: &mockcore.MockCore{
+			Behavior: mockcore.Error,
+		},
+	}
+	tb.modifySettings(rw, req)
+
+	verifyResponseError(t, rw)
+}
+
+func Test_ModifySettings_CoreSucceeds_Success(t *testing.T) {
+	r := &prot.ContainerModifySettings{
+		MessageBase: &prot.MessageBase{
+			ContainerID: "test",
+			ActivityID:  "act",
+		},
+		Request: prot.ResourceModificationRequestResponse{
+			ResourceType: prot.PtMappedDirectory,
+			Settings:     &prot.MappedDirectory{}, // Default values.
+		},
+	}
+
+	req := createRequest(t, prot.ComputeSystemModifySettingsV1, r)
+	rw := createResponseWriter(req)
+
+	tb := &Bridge{
+		coreint: &mockcore.MockCore{
+			Behavior: mockcore.Success,
+		},
+	}
+	tb.modifySettings(rw, req)
+
+	verifyResponseSuccess(t, rw)
+}
```

Comment:
I think my main suggestion for the tests is that they test that the correct values were unpacked into mockcore inconsistently. I think it would be a good idea to add checks that the values were properly unmarshaled into the success tests (e.g. `Test_ModifySettings_CoreSucceeds_Success`), so that we have validation that the correct values make it from the message to the Core.

Reply:
For the success cases I will check all of the params. Working on the fix.

Reply:
Also, sorry, *consistently :)

Reply:
And thanks! :)